src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,245 @@
+package frc.robot.subsystems.shooter;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.CANSparkFlex;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+
+import frc.robot.Constants;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.Velocity;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.Distance;
+import static edu.wpi.first.units.Units.*;
+
+public class Shooter extends SubsystemBase {
+    private CANSparkMax m_feederMotor;
+    private CANSparkFlex m_kickerMotor;
+    private CANSparkFlex m_shooterLeftMotor;
+    private CANSparkFlex m_shooterRightMotor;
+
+    private AbsoluteEncoder pivotAngle;
+    private RelativeEncoder leftShotSpeed;
+    private RelativeEncoder rightShotSpeed;
+
+    private SparkPIDController m_feederPIDController;
+    private SparkPIDController m_kickerPIDController;
+    private SparkPIDController m_shooterLeftPIDController;
+    private SparkPIDController m_shooterRightPIDController;
+
+    // desiredPivot should always be in degrees from the horizontal plane.
+    private double desiredPivot;
+    private Measure<Velocity<Distance>> desiredShotSpeed;
+    private Measure<Velocity<Angle>> desiredRotation;
+
+    public Shooter() {
+        m_feederMotor = new CANSparkMax(Constants.ShooterConstants.feederMotorCANID, MotorType.kBrushless);
+        m_kickerMotor = new CANSparkFlex(Constants.ShooterConstants.kickerMotorCANID, MotorType.kBrushless);
+        m_shooterLeftMotor = new CANSparkFlex(Constants.ShooterConstants.shooterLeftCANID, MotorType.kBrushless);
+        m_shooterRightMotor = new CANSparkFlex(Constants.ShooterConstants.shooterRightCANID, MotorType.kBrushless);
+
+        m_feederMotor.restoreFactoryDefaults();
+        m_kickerMotor.restoreFactoryDefaults();
+        m_shooterLeftMotor.restoreFactoryDefaults();
+        m_shooterRightMotor.restoreFactoryDefaults();
+
+        m_shooterLeftPIDController = m_shooterLeftMotor.getPIDController();
+        m_shooterRightPIDController = m_shooterRightMotor.getPIDController();
+        m_kickerPIDController = m_kickerMotor.getPIDController();
+        m_feederPIDController = m_feederMotor.getPIDController();
+
+        rightShotSpeed = m_shooterRightMotor.getEncoder();
+        leftShotSpeed = m_shooterLeftMotor.getEncoder();
+        leftShotSpeed.setVelocityConversionFactor(1 / (120 * Math.PI));
+        rightShotSpeed.setVelocityConversionFactor(1 / (120 * Math.PI));
+        pivotAngle.setPositionConversionFactor(360);
+
+        desiredPivot = 0;
+        desiredShotSpeed = MetersPerSecond.of(0.0);
+        desiredRotation = RadiansPerSecond.of(0.0);
+        // Set PID values
```

Comment:
Remember to call the burn flash method on every motor at the end. Also, since you are restoring defaults, it is important to set the current limit on every motor.